src/extract/transpile/svelte-wrap.js

```diff
@@ -0,0 +1,46 @@
+const tryRequire = require("semver-try-require");
+const _get = require("lodash/get");
+
+const svelte = tryRequire(
+  "svelte",
+  require("../../../package.json").supportedTranspilers.svelte
+);
+
+// eslint-disable-next-line import/order, node/no-unpublished-require
+const svelteCompile = svelte ? require("svelte/compiler") : false;
+
+module.exports = (pTsWrapper) => ({
+  isAvailable: () => svelteCompile !== false,
+  transpile: async (pSource, pTranspileOptions = {}) => {
+    const optionallyCompileTsToJsInSvelte = await svelteCompile.preprocess(
+      pSource,
+      {
+        script: ({ content, attributes }) => {
+          if (attributes.lang !== "ts" || !pTsWrapper.isAvailable()) {
+            return { code: content };
+          }
+          const compiledToTsFromJs = pTsWrapper.transpile(content, {
+            ...pTranspileOptions,
+            tsConfig: {
+              ..._get(pTranspileOptions, "tsConfig", {}),
+              options: {
+                ..._get(pTranspileOptions, "tsConfig.options", {}),
+                importsNotUsedAsValues: "error",
```

Comment:
1. What is the reason `importsNotUsedAsValues` is on `error` instead of one of the other values (e.g. `preserve` (or the default `remove`)? If I see correctly with this value on `error` the typescript compiler will actually throw, while the imports are still perfectly analysable, which dependency-cruiser's goal here.
2. Is there a reason to have `importsNotUsedAsValues` and `jsx` prevail over their counterparts set in the tsConfig?

Reply:
1. The default `remove` will definitely not work, because of the way we use `svelte-preprocessor` and `svelte/compile`. If you try this value and run `npx mocha test test/extract/transpile/svelte-wrap.spec.js`, the test will fail (missing `import "./Header.svelte";`). I did try with `preserve`, and it seems to be no different than `error`, so I have no strong opinion on that, but the reason I originally chose `error` is because I got it from https://www.npmjs.com/package/@tsconfig/svelte

2. `importsNotUsedAsValues` is explain aboved. As for `jsx`, it's because it's assumed that in Svelte, it's "JS in HTML", not "HTML in JS", and thus the code inside `<script>` tag will be valid JS/TS only, not with JSX extension.

Reply:
1. Thanks for the explanation! Between `error` and `preserve`, with dependency-cruiser's use in mind `preserve` seems like a better fit; `error` will make the typescript compiler throw with an error that isn't relevant for dependency analysis.

2. Clear!

Reply:
Ok I will use `preserve`